client/dashboard/me/tax-details/index.tsx

```diff
@@ -1,11 +1,167 @@
+import { useResetSupportInteraction } from '@automattic/help-center/src/hooks/use-reset-support-interaction';
+import {
+	Card,
+	CardBody,
+	__experimentalVStack as VStack,
+	__experimentalHStack as HStack,
+} from '@wordpress/components';
+import { useDispatch as useDataStoreDispatch, useDispatch } from '@wordpress/data';
 import { __ } from '@wordpress/i18n';
+import { useTranslate } from 'i18n-calypso';
+import { useCallback } from 'react';
+import { useAnalytics } from '../../app/analytics';
+import { useTaxName } from '../../app/hooks/use-country-list';
+import { useGeoLocationQuery } from '../../app/queries/geolocation';
+import InlineSupportLink from '../../components/inline-support-link';
 import { PageHeader } from '../../components/page-header';
 import PageLayout from '../../components/page-layout';
+import { localizeUrl } from './localize-url';
+import useRecordUserTaxEvents from './use-record-user-tax-events';
+import useUserTaxDetails from './use-user-tax-details';
+import UserTaxForm from './user-tax-form';
+import './style.scss';
+
+const HELP_CENTER_STORE = 'automattic/help-center';
+
+export default function UserTaxInfoPage() {
+	const { recordTracksEvent } = useAnalytics();
+	const translate = useTranslate();
+	const { data: geoData } = useGeoLocationQuery();
+	const { fetchError, userTaxDetails } = useUserTaxDetails();
```

Comment:
Data like this, where the view isn't really usable until this data has been loaded, should be loaded in the render. Here's an example:

https://github.com/Automattic/wp-calypso/blob/e75f49d6168dc0fde8470da6c9d057baec39a39b/client/dashboard/app/router/me.tsx#L68

When the data is fetched in the router it means we can then use React Suspense in the component.

```suggestion
	const { data: geoData } = useSuspenseQuery( geoLocationQuery() );
	const { data: userTaxDetails } = useSuspenseQuery( userTaxDetailsQuery() );
```

And we know we don't have to deal with the data being undefined.

Loading the data for each view in the router means that we will be able to have an interface that isn't full of as many loading spinners 🤞

Reply:
I think I made this change in 4c801ee. If not, it would have been in another commit, but this change has been made, maybe 1fbf5a0.